Training/Controller/Note/Save.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Codifi_Training
+ *
+ * @copyright   Copyright (c) 2021 Codifi
+ * @author      Pavel Zelenevich <pzelenevich@codifi.me>
+ */
+
+namespace Codifi\Training\Controller\Note;
+
+use Magento\Framework\App\Action\Action;
+use Magento\Framework\App\Action\Context;
+use Magento\Framework\App\ResponseInterface;
+use Codifi\Training\Model\CustomerNoteFactory;
+use Codifi\Training\Model\ResourceModel\CustomerNote as CustomerNoteResource;
+use Magento\Framework\Controller\Result\JsonFactory;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Controller\Result\Json;
+use Magento\Framework\Controller\ResultInterface;
+
+/**
+ * Class Save
+ * @package Codifi\Training\Controller\Note
+ */
+class Save extends Action
+{
+    /**
+     * Customer note factory.
+     *
+     * @var CustomerNoteFactory
+     */
+    private $customerNoteFactory;
+
+    /**
+     * Customer note resource model.
+     *
+     * @var CustomerNoteResource
+     */
+    private $customerNoteResource;
+
+    /**
+     * Json factory.
+     *
+     * @var JsonFactory
+     */
+    private $jsonFactory;
+
+    /**
+     * Save constructor.
+     *
+     * @param Context $context
+     * @param JsonFactory $jsonFactory
+     * @param CustomerNoteFactory $customerNoteFactory
+     * @param CustomerNoteResource $customerNoteResource
+     */
+    public function __construct(
+        Context $context,
+        JsonFactory $jsonFactory,
+        CustomerNoteFactory $customerNoteFactory,
+        CustomerNoteResource $customerNoteResource
+    ) {
+        parent::__construct($context);
+        $this->jsonFactory = $jsonFactory;
+        $this->customerNoteFactory = $customerNoteFactory;
+        $this->customerNoteResource = $customerNoteResource;
+    }
+
+    /**
+     * Execute function.
+     *
+     * @return ResponseInterface|Json|ResultInterface
+     * @throws \Exception
+     */
+    public function execute()
+    {
+        $note = $this->getRequest()->getParam('note');
+        $customerId = $this->getRequest()->getParam('customer_id');
+
+        $customerNoteModel = $this->customerNoteFactory->create();
+        $resultJson = $this->jsonFactory->create();
+
+        if ($note) {
```

Comment:
`customer_id` can't be empty so you should add additional validation and error processing for this case